libpromises/evalfunction.c

```diff
@@ -4209,6 +4209,23 @@ static FnCallResult FnCallShuffle(EvalContext *ctx, ARG_UNUSED const Policy *pol
     return (FnCallResult) { FNCALL_SUCCESS, (Rval) { shuffled, RVAL_TYPE_LIST } };
 }
 
+static FnCallResult FnCallInt(ARG_UNUSED EvalContext *ctx, ARG_UNUSED const Policy *policy, ARG_UNUSED const FnCall *fp, const Rlist *finalargs)
+{
+    if(finalargs == NULL)
+    {
+        return FnFailure();
+    }
+
+    char *str = RlistScalarValueSafe(finalargs);
+
+    double val;
+    bool ok = DoubleFromString(str, &val);
+    if(!ok) {
+        // Log from DoubleFromString
+        return FnFailure();
+    }
+    return FnReturnF("%ld", (long) val);  // Discard decimals
```

Comment:
I'd personally do `FnReturnF("%jd", (intmax_t) val);` to avoid the riddle of "which integer type is big enough to hold an integer part of a `double`".